feat(Form Component): handle boolean types & adjust migrations per mark #45

Abdulmajeed-Jamaan opened the pull request.

Abdulmajeed-Jamaan committed:
feat(Form Component): refactor to handle boolean types

Abdulmajeed-Jamaan committed:
feat(migrations): set the proper value column type per mark

atmonshi commented on src/Forms/Concerns/HasColors.php:
```diff
@@ -2,46 +2,28 @@
-     * @var array<string>|Arrayable<array-key, string>|Closure|null
+     * @var array<string|int, string>|string|null
      */
-    protected array | Arrayable | Closure | null $colors = null;
```
these were from filament, to account for any scenario.
I think allowing Closures is a good idea

atmonshi commented on src/Forms/Concerns/HasSelectableIcons.php:
```diff
@@ -2,58 +2,44 @@
-     * @var array<string|int|bool, string>|Closure|null
+     * @var array<string|int, string>|string
      */
-    protected array | Closure | null $selectedIcons = null;
```
I think also better to allow at least Closure

atmonshi reviewed (changes requested):
notes

Abdulmajeed-Jamaan commented:
> notes

Understandable, but do you have  in mind any real use case for it ?

I prefer to handle use cases as needed, and we can handle closures anytime later without breaking changes.

atmonshi commented:
> Understandable, but do you have in mind any real use case for it ?

set the color based on the current tenant from its profile!

dependabot[bot] committed:
chore(deps): bump stefanzweifel/git-auto-commit-action from 5 to 6

Bumps [stefanzweifel/git-auto-commit-action](https://github.com/stefanzweifel/git-auto-commit-action) from 5 to 6.
- [Release notes](https://github.com/stefanzweifel/git-auto-commit-action/releases)
- [Changelog](https://github.com/stefanzweifel/git-auto-commit-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/stefanzweifel/git-auto-commit-action/compare/v5...v6)

---
updated-dependencies:
- dependency-name: stefanzweifel/git-auto-commit-action
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Abdulmajeed-Jamaan committed:
feat(colors & icons): handle Closure and Arrayable parameters types

atmonshi committed:
Merge pull request #48 from lara-zeus/dependabot/github_actions/stefanzweifel/git-auto-commit-action-6

chore(deps): bump stefanzweifel/git-auto-commit-action from 5 to 6

Abdulmajeed-Jamaan committed:
feat: create mark blade component

Abdulmajeed-Jamaan committed:
refactor: use mark blade component

dependabot[bot] committed:
chore(deps): bump aglipanci/laravel-pint-action from 2.5 to 2.6

Bumps [aglipanci/laravel-pint-action](https://github.com/aglipanci/laravel-pint-action) from 2.5 to 2.6.
- [Release notes](https://github.com/aglipanci/laravel-pint-action/releases)
- [Commits](https://github.com/aglipanci/laravel-pint-action/compare/2.5...2.6)

---
updated-dependencies:
- dependency-name: aglipanci/laravel-pint-action
  dependency-version: '2.6'
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #49 from lara-zeus/dependabot/github_actions/aglipanci/laravel-pint-action-2.6

chore(deps): bump aglipanci/laravel-pint-action from 2.5 to 2.6

dependabot[bot] committed:
chore(deps): bump actions/checkout from 4 to 5

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Abdulmajeed-Jamaan committed:
W.I.P

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
Update composer.json

Abdulmajeed-Jamaan committed:
Merge branch '40-filament-form-component' of https://github.com/lara-zeus/mark into 40-filament-form-component

Abdulmajeed-Jamaan committed:
Direct redirect to the component

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
chore: move styles from inline html to css file

Abdulmajeed-Jamaan committed:
Merge branch '40-filament-form-component' of https://github.com/lara-zeus/mark into 40-filament-form-component

Abdulmajeed-Jamaan committed:
fix: make models unguarded by default for the demo

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
move tests to demo app

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
move larastan and pint to the demo as well

larastan require laravel context and we remove testbench previously, so i had to move it, and did pint as well

Abdulmajeed-Jamaan committed:
reformat

Abdulmajeed-Jamaan committed:
chore: setup after install as well

Abdulmajeed-Jamaan committed:
chore: alias dev script

Abdulmajeed-Jamaan committed:
chore: add setup script alias

Abdulmajeed-Jamaan committed:
refactor: scale class condition

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
Table component done

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
composer lint

Abdulmajeed-Jamaan committed:
Merge branch '40-filament-form-component' of https://github.com/lara-zeus/mark into 40-filament-form-component

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
handle readonly state for infolist

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
Update mark.blade.php

Abdulmajeed-Jamaan committed:
feat: add infolists component

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
fix

Abdulmajeed-Jamaan committed:
fix phpstan workflow

Abdulmajeed-Jamaan committed:
Update phpstan.yml

Abdulmajeed-Jamaan committed:
Update composer.json

Abdulmajeed-Jamaan committed:
Update phpstan.yml

Abdulmajeed-Jamaan committed:
Update phpstan.yml

Abdulmajeed-Jamaan committed:
Update phpstan.yml

Abdulmajeed-Jamaan committed:
Update phpstan.yml

Abdulmajeed-Jamaan committed:
Update phpstan.yml

Abdulmajeed-Jamaan committed:
Update phpstan.yml

Abdulmajeed-Jamaan committed:
fix

Abdulmajeed-Jamaan committed:
Merge pull request #50 from lara-zeus/dependabot/github_actions/actions/checkout-5

chore(deps): bump actions/checkout from 4 to 5

Abdulmajeed-Jamaan committed:
Merge branch '1.x' into 40-filament-form-component

Abdulmajeed-Jamaan committed:
ci: workflows naming and commands

Abdulmajeed-Jamaan committed:
style: run laravel pint

Abdulmajeed-Jamaan committed:
fix: update git ignore

Abdulmajeed-Jamaan committed:
Update pest.yml

Abdulmajeed-Jamaan committed:
Update pest.yml

Abdulmajeed-Jamaan committed:
ci: fix test workflow

Abdulmajeed-Jamaan committed:
fix: handle bookmark breaking changes with docs

Abdulmajeed-Jamaan committed:
style: fix wording

Abdulmajeed-Jamaan committed:
Update CommentResource.php

Abdulmajeed-Jamaan committed:
style: run laravel pint

Abdulmajeed-Jamaan committed:
Update demo assets and package lock

Update demo frontend: adjust Filament forms CSS and the markdown editor JS, and refresh demo/package-lock.json to reflect dependency resolution after the frontend changes.
